root_orchestrator/resource-abstractor/db/clusters_db.py

```diff
@@ -1,26 +1,88 @@
+from datetime import datetime
+
 import db.mongodb_client as db
 from bson.objectid import ObjectId
+from db.clusters_helper import get_freshness_threshold
+
+
+def create_cluster(data):
+    inserted = db.mongo_clusters.insert_one(data)
+
+    return db.mongo_clusters.find_one({"_id": inserted.inserted_id})
 
-from .clusters_helper import get_freshness_threshhold
+
+def update_cluster(cluster_id, data):
+    return db.mongo_clusters.find_one_and_update(
+        {"_id": ObjectId(cluster_id)},
+        {"$set": data},
+        return_document=True,
+    )
 
 
 def find_clusters(filter):
     pipeline = [
         {"$match": filter},
         {
             "$addFields": {
-                "active": {"$gt": ["$last_modified_timestamp", get_freshness_threshhold()]}
+                "active": {"$gt": ["$last_modified_timestamp", get_freshness_threshold()]}
             }
         },
     ]
-    return db.mongo_clusters.db.clusters.aggregate(pipeline)
+    return db.mongo_clusters.aggregate(pipeline)
 
 
 def find_cluster_by_id(cluster_id):
     cluster = list(find_clusters({"_id": ObjectId(cluster_id)}))
     return cluster[0] if cluster else None
 
 
-def find_cluster_by_name(name):
-    cluster = list(find_clusters({"cluster_name": name}))
-    return cluster[0] if cluster else None
+def find_cluster_by_name(cluster_name):
+    return db.mongo_clusters.find_one({"cluster_name": cluster_name})
+
+
+def update_cluster_information(cluster_id, data):
+    """Save aggregated Cluster Information"""
+
+    datetime_now = datetime.now()
+    datetime_now_timestamp = datetime.timestamp(datetime_now)
+
+    cpu_percent = data.get("cpu_percent")
```

Comment:
Idea. Can we iterate through the keywords instead of manually setting them?, maybe define a translation map for the pre-defined known keywords such as cpu_cores. This way if a cluster was extended with additional capabilities such as... quantum cores, news sensors, or other stuff, we don't need to extend the abstractor. What do you think?

Reply:
This could definitely be cleaned up. However we would still need to extend the translation map when new values are added, or perhaps I am missing something.